pkg/reconciler/identity/reconciler.go

```diff
@@ -165,7 +166,9 @@ func setIamPolicy(ctx context.Context, action, projectID string, gServiceAccount
 		return fmt.Errorf("failed to get project id: %w", err)
 	}
 
-	resource := fmt.Sprintf("projects/%s/serviceAccounts/%s", projectId, gServiceAccount)
+	// Extract gServiceAccount's project name.
+	gsaProject := strings.Split(strings.Split(gServiceAccount, "@")[1], ".")[0]
```

Comment:
Can we add an example service account name here, so it's clearer how the project name is extracted?